assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -7044,31 +7044,29 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
             // the grade point
             String gradePoints = params.getString(NEW_ASSIGNMENT_GRADE_POINTS);
             state.setAttribute(NEW_ASSIGNMENT_GRADE_POINTS, gradePoints);
-            if (gradePoints != null) {
-                if (gradeType == SCORE_GRADE_TYPE) {
-                    if ((gradePoints.length() == 0)) {
-                        // in case of point grade assignment, user must specify maximum grade point
-                        addAlert(state, rb.getString("plespethe3"));
-                    } else {
-                        Integer scaleFactor = assignmentService.getScaleFactor();
-                        try {
-                            if (StringUtils.isNotEmpty(assignmentRef)) {
-                                Assignment assignment = assignmentService.getAssignment(assignmentId);
-                                if (assignment != null) {
-                                    scaleFactor = assignment.getScaleFactor();
-                                }
-                            }
-                        } catch (IdUnusedException | PermissionException e) {
-                            log.error(e.getMessage());
-                        }
+			if (gradePoints != null && gradeType == SCORE_GRADE_TYPE) {
+				if ((gradePoints.length() == 0)) {
+					// in case of point grade assignment, user must specify maximum grade point
+					addAlert(state, rb.getString("plespethe3"));
+				} else {
+					Integer scaleFactor = assignmentService.getScaleFactor();
+					try {
+						if (StringUtils.isNotEmpty(assignmentRef)) {
+							Assignment assignment = assignmentService.getAssignment(assignmentId);
+							if (assignment != null && assignment.getScaleFactor() != null) {
```

Comment:
if the scale factor on the assignment is null should use:
org.sakaiproject.assignment.impl.AssignmentServiceImpl#getScaleFactor

Reply:
Before the try block (line 7052) we are already getting the scaleFactor from assignmentService.

Reply:
ok looks good